File: app/src/main/java/com/linktalent/app/Model/Entity/Media.java
package com.linktalent.app.Model.Entity;

import com.linktalent.app.Model.Enums.MediaType;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;


@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public final class Media {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(unique = true)
    private String url;

    @NotBlank(message = "description cannot be blank")
    private String description;

    @Enumerated(EnumType.STRING)
    private MediaType type;

    @ManyToOne
    private Post post;

    @ManyToOne
    private Sport sport;
}

File: frontend/src/Types/Media.ts
import { UUID } from "crypto";

export type Media = {
    id: number;
    url: string;
    description: string;
    type: string;
    post: object;
    sport: object;
}

export type MediaRequest = {
    id?: number;
    url: string;
    description: string;
    type: string;
    post: UUID;
    sport: UUID;
}
